include/deal.II/matrix_free/operators.h

```diff
@@ -678,13 +686,18 @@ namespace MatrixFreeOperators
   void
   Base<dim,Number>::
   initialize (const MatrixFree<dim,Number> &data_,
-              const MGConstrainedDoFs &mg_constrained_dofs,
-              const unsigned int level)
+              const MGConstrainedDoFs      &mg_constrained_dofs,
+              const unsigned int            level)
   {
     AssertThrow (level != numbers::invalid_unsigned_int,
                  ExcMessage("level is not set"));
+    if (data_.n_macro_cells() > 0)
+      {
+        AssertDimension(static_cast<int>(level),
```

Comment:
I also thought about removing the level but decided that it is less clear without it. So we seem to agree on this topic! :+1: